server/jetstream_cluster.go

```diff
@@ -767,12 +768,55 @@ func (js *jetStream) monitorCluster() {
 			}
 		case isLeader = <-lch:
 			js.processLeaderChange(isLeader)
+			if isLeader && !beenLeader {
+				beenLeader = true
+				if n.NeedSnapshot() {
+					if err := n.InstallSnapshot(js.metaSnapshot()); err != nil {
+						s.Warnf("Error snapshotting JetStream cluster state: %v", err)
+					}
+				}
+				js.checkClusterSize()
+			}
 		case <-t.C:
 			doSnapshot()
 		}
 	}
 }
 
+// This is called on first leader transition to double check the peers and cluster set size.
+func (js *jetStream) checkClusterSize() {
+	s, n := js.server(), js.getMetaGroup()
+	if n == nil || !n.Leader() {
```

Comment:
The check of `n.Leader()` bothers me. We are called because we became leader. Having the test here means to me that the state can change underneath us.. if that is the case, then what's the validity of this call if the state suddenly change after this check?

Reply:
Will remove.